ReindexFailureTests fails with in-flight search context

ReindexFailureTests suite fails quite often today (see the [stats](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-1y,mode:quick,to:now))&_a=(columns:!(_source),filters:!(),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'stacktrace:%22There%20are%20still%20%5B1%5D%20in-flight%20contexts%22'),sort:!(time,desc)))). I could not reproduce this locally.

```
java.lang.AssertionError: There are still [1] in-flight contexts. The first one's creation site is listed as the cause of this exception.
	at org.elasticsearch.search.MockSearchService.assertNoInFlightContext(MockSearchService.java:51)
	at org.elasticsearch.test.ESTestCase.lambda$ensureAllSearchContextsReleased$6(ESTestCase.java:507)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:866)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:840)
	at org.elasticsearch.test.ESTestCase.ensureAllSearchContextsReleased(ESTestCase.java:507)
	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:1967)
```

Some instances:
- https://gradle-enterprise.elastic.co/s/p4cnifsgjc67s
- https://gradle-enterprise.elastic.co/s/7vtrpwj2lqg3y

--- a/test/framework/src/main/java/org/elasticsearch/search/MockSearchService.java
+++ b/test/framework/src/main/java/org/elasticsearch/search/MockSearchService.java
@@ -74,8 +74,8 @@ public class MockSearchService extends SearchService {
 
     @Override
     protected void putContext(SearchContext context) {
-        super.putContext(context);
         addActiveContext(context);
+        super.putContext(context);
     }
 
     @Override

